lib/sidekiq/redis_connection.rb

```diff
@@ -115,27 +117,20 @@ def determine_redis_provider
         # REDIS_PROVIDER=MY_REDIS_URL
         # and Sidekiq will find your custom URL variable with no custom
         # initialization code at all.
+        #
         p = ENV['REDIS_PROVIDER']
         if p && p =~ /\:/
-          Sidekiq.logger.error <<-EOM
+          raise <<-EOM
```

Comment:
Since you now require Ruby 2.5+, you could rewrite this using `<<~` and thus indent the text to keep it tidy.